docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,251 @@
+```
+
+#### Application Services
+
+##### [MessageBus]
+
+Similar to above, the Application Services `MessageBus` configuration will change to allow for secure connection to the MessageBus. The `Username` & `Password` have been replaced with `Authmode` & `SecretPath` for secure connections. See **[Secure Connections](#secure-connections)** section below for details.
+
+```toml
+[MessageBus.Optional]
+    # MQTT Specific options
+    # Client Identifiers
+    ClientId ="<app sevice key>"
+    # Connection information
+    Qos          =  "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+    KeepAlive    =  "10" # Seconds (must be 2 or greater)
+    Retained     = "false"
+    AutoReconnect  = "true"
+    ConnectTimeout = "5" # Seconds
+    SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+    Authmode = "none" # Valid values are: `none`, `usernamepassword`, `clientcert` or `cacert`
+    Secretpath = "messagebus"  # Path in secret store used if Authmode not `none`
+```
+
+##### [Binding]
+
+The `Binding` configuration section will require change for the subscribe topics scheme describe in the [MessageBus Topics](#messagebus-topics) section above to filter for Events from specific device profiles or devices. `SubscribeTopic` will change from a string property containing a single topic to the `SubscribeTopics` string property containing a comma separated list of topics. This allows for the flexibility for the property to be a single topic with the `#` wild card so the Application Service receives all Events as it does today.
+
+Receive only Events from the `Random-Integer-Device` and `Random-Boolean-Device` profiles
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/Random-Integer-Device, edgex/events/Random-Boolean-Device"
+```
+Receive only Events from the  `Random-Integer-Device1` from the `Random-Integer-Device` profile
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/Random-Integer-Device/Random-Integer-Device1"
+```
+
+or receives all Events:
+
+```toml
+[Binding]
+Type="messagebus"
+SubscribeTopics="edgex/events/#"
+```
+
+### Secure Connections
+
+As stated earlier,  this ADR is dependent on the  [**Secret Provider for All**](TBD) ADR to provide a common Secret Provider for all Edgex Services to access their secrets. Once this is available, the MessageBus connection can be secured via the following configurable authentications modes which follows the implementation for secure MQTT Export and secure MQTT Trigger used in Application Services.
+
+- **none** - No authentication used
+- **usernamepassword** - Username & password authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **clientcert** - Client certificate and key for authentication. CA Cert used in conjunction if it exists in the Secret Provider. 
+- **cacert** - CA Certificate only, i.e. just a TLS enabled connection
```

Comment:
Then why not call this "server" or "tls"? AFAIK a CA certificate is only required if the broker's TLS certificate was signed by a CA unknown to the system (or self-signed?). If the broker's TLS cert was signed by a CA known to the system, then there's nothing additional to configure. So in my mind "cacert" is an optional security parameter, not an actual mode.

@bnevis-i ^^

Reply:
@tonyespy , would it help to change setting name from `Authmode` to `SecureMode` since we are not just dealing with auth?

Reply:
Actually after giving it some thought, I think a better idea would be to call this `ClientAuth` and only support:

  * **none**
  * **username/password** (aka basic)
  * **clientcert**

Whether or not TLS (i.e. server cert) is used is actually orthagonal...

Reply:
Sounds good to me.

Reply:
Works for me